org.gajim.Gajim.json

```diff
@@ -0,0 +1,381 @@
+{
+    "app-id": "org.gajim.Gajim",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.28",
+    "sdk": "org.gnome.Sdk",
+    "command": "gajim",
+    "finish-args": [
+        "--share=ipc",
+        "--share=network",
+        "--socket=x11",
+        "--socket=wayland",
+        "--socket=pulseaudio",
+        "--filesystem=home",
```

Comment:
What do you need home access for? You should be able to select or download files using the portal if you use GTKNativeFileChooser

Reply:
We inherit from Gtk.FileChooserDialog, which seems unsupported by the portal because I tried without `--filesystem=home` and it didn't work.

Reply:
Yes, you need to use https://developer.gnome.org/gtk3/stable/gtk3-GtkFileChooserNative.html instead in order to support sandboxing. It's a pretty simple change.

Reply:
Is this a blocking issue?
Because it seems that it's not that simple in this case, but we will try to fix the issue upstream.

Reply:
In most cases it is extremely trivial to port to this API. The only time it isn't is when you make custom dialogs.

Reply:
@lovetox when you're inside a flatpak container there's a file at /.flatpak-info that has a bunch of information about the environment. Here's an example from my system.

```nick@endless-nedr:~/Projects/flathub$ flatpak run --command=bash org.freedesktop.Bustle 
nick@endless-nedr:~/Projects/flathub$ cat /.flatpak-info 
[Application]
name=org.freedesktop.Bustle
runtime=runtime/org.gnome.Platform/x86_64/3.26

[Instance]
app-path=/var/lib/flatpak/app/org.freedesktop.Bustle/x86_64/stable/868dc6c3bf8444e2f8be8bfb6783feccca91989caf0730ded60fb0b3a46b0ff2/files
app-commit=868dc6c3bf8444e2f8be8bfb6783feccca91989caf0730ded60fb0b3a46b0ff2
runtime-path=/var/lib/flatpak/runtime/org.gnome.Platform/x86_64/3.26/d368d1e6078d9c8f51ccbdb1b0aae875012b28fab8e2ba844ad347e46098ecaf/files
runtime-commit=d368d1e6078d9c8f51ccbdb1b0aae875012b28fab8e2ba844ad347e46098ecaf
runtime-extensions=org.freedesktop.Platform.GL.default=local;org.freedesktop.Platform.GL.default=local;org.freedesktop.Platform.Icontheme.EndlessOS=0090069a06a68551d5d6ec1569f7dd8be3c58cdeb03f6857b3701f3f6caa2f8b;org.freedesktop.Platform.VAAPI.Intel=818a1504fad164f9a975b8753dcf0abf0b46db02ce06210b825dfa15ec8af4f7;org.freedesktop.Platform.ffmpeg=b5f77229b85f1cfb266562b26c3b39221a3db71ea01cf9d2e53d54dbf97578d1;org.freedesktop.Sdk.Extension.gfortran-62=5b89f0da9d3540dde123f98a04235b630fa35122175b75c037874c6857a10000;org.freedesktop.Sdk.Extension.openjdk9=d5345ae755761bc6571c7b2f55aa685c55b475af7260c63e1fd6f7353299d311;org.freedesktop.Sdk.Extension.rust-stable=184292e28334986c745117342ac2b83bce1af862aac3b13460eadc071076dc51;org.gnome.Platform.Locale=dab9f8e991a1225995d0160560c722783bb9003ea01e108d02bee18cf47682ab
branch=stable
flatpak-version=0.10.1

[Context]
shared=ipc;
sockets=x11;wayland;session-bus;system-bus;
filesystems=host;

[Session Bus Policy]
org.freedesktop.portal.*=talk

[Environment]
GST_PLUGIN_SYSTEM_PATH=/app/lib/gstreamer-1.0:/usr/lib/extensions/gstreamer-1.0:/usr/lib/gstreamer-1.0
GI_TYPELIB_PATH=/app/lib/girepository-1.0
XDG_DATA_DIRS=/app/share:/usr/share:/usr/share/runtime/share:/run/host/share
```